core/unit_test/view/TestExtentsDatatypeConversion.cpp

```diff
@@ -0,0 +1,70 @@
+#include <Kokkos_Core.hpp>
+#include <type_traits>
+
+#ifdef KOKKOS_ENABLE_IMPL_MDSPAN
+
+namespace {
+// Conversion from DataType to extents
+// 0-rank view
+static_assert(std::is_same_v<typename Kokkos::Experimental::Impl::
+                                 ExtentsFromDataType<std::size_t, double>::type,
+                             std::experimental::extents<std::size_t>>);
+// Only dynamic
+static_assert(
+    std::is_same_v<typename Kokkos::Experimental::Impl::ExtentsFromDataType<
+                       std::size_t, double***>::type,
+                   std::experimental::extents<
+                       std::size_t, std::experimental::dynamic_extent,
+                       std::experimental::dynamic_extent,
+                       std::experimental::dynamic_extent>>);
+// Only static
+static_assert(std::is_same_v<
+              typename Kokkos::Experimental::Impl::ExtentsFromDataType<
+                  std::size_t, double[2][3][17]>::type,
+              std::experimental::extents<std::size_t, std::size_t{2},
+                                         std::size_t{3}, std::size_t{17}>>);
+// Both dynamic and static
+static_assert(
+    std::is_same_v<typename Kokkos::Experimental::Impl::ExtentsFromDataType<
+                       std::size_t, double* * [3][2][8]>::type,
+                   std::experimental::extents<
+                       std::size_t, std::experimental::dynamic_extent,
+                       std::experimental::dynamic_extent, std::size_t{3},
+                       std::size_t{2}, std::size_t{8}>>);
+
+// Conversion from extents to DataType
+// 0-rank extents
+static_assert(
+    std::is_same_v<double,
+                   typename Kokkos::Experimental::Impl::DataTypeFromExtents<
+                       double, std::experimental::extents<std::size_t>>::type>);
```

Comment:
Introduce a macro or a trait so the test read something
```C++
CHECK_DATA_TYPE_TO_EXTENT(double, extents<std::size_t>);
CHECK_DATA_TYPE_TO_EXTENT(double***, dextents<std::size_t, 3>);
```
essentially eliminates all the boiler plate code and let the reader focus on what is being mapped to what

Reply:
I used a helper template instead of a macro, hopefully it's clearer.